src/Explorer/Tabs/DocumentsTabV2/DocumentsTableComponent.tsx

```diff
@@ -121,24 +123,58 @@ export const DocumentsTableComponent: React.FC<IDocumentsTableComponentProps> =
     [columnHeaders],
   );
 
-  const onIdClicked = useCallback((index: number) => onSelectedRowsChange(new Set([index])), [onSelectedRowsChange]);
+  const [selectionStartIndex, setSelectionStartIndex] = React.useState<number>(undefined);
+  const onTableCellClicked = useCallback(
+    (e: React.MouseEvent, index: number) => {
+      const result = selectionHelper(selectedRows as Set<number>, index, e.shiftKey, e.ctrlKey, selectionStartIndex);
```

Comment:
For MacOS it should be `e.metaKey` instead of `e.ctrlKey` if I'm not mistaken.

Reply:
That is correct. Turns out we already had code to detect platform and check ctrl vs meta for Tables API which I moved to a more common place.